lib/_http_outgoing.js

```diff
@@ -345,10 +353,20 @@ OutgoingMessage.prototype._send = function _send(data, encoding, callback) {
       this._onPendingData(header.length);
     }
     this._headerSent = true;
+    this[kContentLength] = _getContentLength.call(this);
   }
   return this._writeRaw(data, encoding, callback);
 };
 
+function _getContentLength() {
+  const outgoing = this.getHeader('content-length');
```

Comment:
We can do this more efficiently by simply checking:

```js
!this._removedContLen && typeof this._contentLength === 'number'
```

and then using `this._contentLength`.